File: user-service/src/main/java/rs/edu/raf/service/FavouriteRecipientService.java
package rs.edu.raf.service;

import rs.edu.raf.dto.FavouriteRecipientDTO;
import rs.edu.raf.dto.FavouriteRecipientRequestDTO;

import java.util.List;

/**
 * This interface defines methods for managing favorite users.
 */
public interface FavouriteRecipientService {

    /**
     * Adds a favorite user.
     *
     * @param favouriteRecipientRequestDTO The DTO containing information about the favorite user to add.
     * @return The added {@link FavouriteRecipientDTO} object.
     */
    FavouriteRecipientDTO addFavouriteClient(FavouriteRecipientRequestDTO favouriteRecipientRequestDTO);

    /**
     * Edits a favorite user.
     *
     * @param favouriteRecipientRequestDTO The DTO containing updated information about the favorite user.
     * @return The edited {@link FavouriteRecipientDTO} object.
     */
    FavouriteRecipientDTO editFavouriteClient(FavouriteRecipientRequestDTO favouriteRecipientRequestDTO);

    /**
     * Deletes a favorite user by ID.
     *
     * @param id The ID of the favorite user to delete.
     */
    void deleteFavouriteClient(Long id);

    /**
     * Finds favorite users for currently logged user.
     *
     * @return A list of {@link FavouriteRecipientDTO} objects representing the favorite users of the specified user.
     */
    List<FavouriteRecipientDTO> findFavouriteClientsForClient();

}

File: user-service-kotlin/src/main/kotlin/org/example/userservicekotlin/service/FavouriteRecipientService.kt
package org.example.userservicekotlin.service

import org.example.userservicekotlin.dto.FavouriteRecipientCreateDTO
import org.example.userservicekotlin.dto.FavouriteRecipientDTO

interface FavouriteRecipientService {
    /**
     * Adds a favorite user.
     *
     * @param favouriteRecipientCreateDTO The DTO containing information about the favorite user to add.
     * @return The added [FavouriteRecipientDTO] object.
     */
    fun addFavouriteClient(favouriteRecipientCreateDTO: FavouriteRecipientCreateDTO): FavouriteRecipientDTO

    /**
     * Edits a favorite user.
     *
     * @param favouriteRecipientCreateDTO The DTO containing updated information about the favorite user.
     * @return The edited [FavouriteRecipientDTO] object.
     */
    fun editFavouriteClient(favouriteRecipientCreateDTO: FavouriteRecipientCreateDTO): FavouriteRecipientDTO?

    /**
     * Deletes a favorite user by ID.
     *
     * @param id The ID of the favorite user to delete.
     */
    fun deleteFavouriteClient(id: Long?)

    /**
     * Finds favorite users for currently logged user.
     *
     * @return A list of [FavouriteRecipientDTO] objects representing the favorite users of the specified user.
     */
    fun findFavouriteClientsForClient(): List<FavouriteRecipientDTO>
}